File: docs/vsto/codesnippet/VisualBasic/trin_word_document_menus.vb/thisdocument.vb
Public Class ThisDocument

    '<Snippet2>
    Public selectedBookmark As Microsoft.Office.Tools.Word.Bookmark
    Public ShowBoldButton As String = False
    Public ShowItalicButton As String = False
    Public WordTrue As Integer = -1
    Public WordFalse As Integer = 0
    '</Snippet2>

    Private Sub ThisDocument_Startup() Handles Me.Startup

    End Sub

    Private Sub ThisDocument_Shutdown() Handles Me.Shutdown

    End Sub

    '<Snippet3>

    Private Sub Bookmark_BeforeRightClick(ByVal sender As Object, ByVal e _
    As Microsoft.Office.Tools.Word.ClickEventArgs) _
    Handles Bookmark1.BeforeRightClick, Bookmark2.BeforeRightClick
        Dim startPosition As Integer = 0
        Dim i As Integer

        ' If bookmarks overlap, get bookmark closest to cursor.
        For i = 1 To e.Selection.Bookmarks.Count
            If e.Selection.Bookmarks(i).Start > startPosition Then
                startPosition = e.Selection.Bookmarks(i).Start
            End If
        Next

        ' If closest bookmark is the sender, show the shortcut menu options.
        If DirectCast(sender, Microsoft.Office.Tools.Word.Bookmark).Start = _
            startPosition Then
            selectedBookmark = DirectCast(sender, Microsoft.Office.Tools.Word.Bookmark)

            ShowBoldButton = True
            ShowItalicButton = True

        End If

    End Sub
    '</Snippet3>

    '<Snippet1>
    Protected Overrides Function CreateRibbonExtensibilityObject() As Microsoft.Office.Core.IRibbonExtensibility
        Return New Ribbon1()
    End Function
    '</Snippet1>
End Class

File: docs/vsto/codesnippet/CSharp/trin_word_document_menus.cs/thisdocument.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Xml.Linq;
using Microsoft.Office.Tools.Word;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Office = Microsoft.Office.Core;
using Word = Microsoft.Office.Interop.Word;

namespace CS
{
    public partial class ThisDocument
    {
        //<Snippet2>
        public Microsoft.Office.Tools.Word.Bookmark selectedBookmark;
        public bool showBoldButton = false;
        public bool showItalicButton = false;
        public int WordTrue = -1;
        public int WordFalse = 0;

        //</Snippet2>

        //<Snippet4>
        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            this.bookmark1.BeforeRightClick += new ClickEventHandler(bookmark_BeforeRightClick);
            this.bookmark2.BeforeRightClick += new ClickEventHandler(bookmark_BeforeRightClick);
        }
        //</Snippet4>

        //<Snippet3>

        void bookmark_BeforeRightClick(object sender, ClickEventArgs e)
        {
            int startPosition = 0;

            // If bookmarks overlap, get bookmark closest to cursor.
            for (int i = 1; i <= e.Selection.Bookmarks.Count; i++)
            {
                if (e.Selection.Bookmarks[i].Start > startPosition)
                {
                    startPosition = e.Selection.Bookmarks[i].Start;
                }
            }

            // If closest bookmark is the sender, show shortcut menu options.
            if (((Microsoft.Office.Tools.Word.Bookmark)sender).Start == startPosition)
            {
                selectedBookmark = (Microsoft.Office.Tools.Word.Bookmark)sender;
                
                showBoldButton = true;
                showItalicButton = true;

            }

        }

        //</Snippet3>
        //<Snippet1>
        protected override Microsoft.Office.Core.IRibbonExtensibility CreateRibbonExtensibilityObject()
        {
            return new Ribbon1();
        }
        //</Snippet1>
        


        private void ThisDocument_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisDocument_Startup);
            this.Shutdown += new System.EventHandler(ThisDocument_Shutdown);
        }

        #endregion
    }
}
